pkg/scheduler/scheduler.go

```diff
@@ -461,9 +462,16 @@ func (s *Scheduler) QuerierLoop(querier schedulerpb.SchedulerForQuerier_QuerierL
 
 		r := req.(*schedulerRequest)
 
-		s.queueDuration.Observe(time.Since(r.enqueueTime).Seconds())
+		reqEnqueueTime := time.Since(r.enqueueTime)
+		s.queueDuration.Observe(reqEnqueueTime.Seconds())
 		r.queueSpan.Finish()
 
+		// Add HTTP header to the request containing the query enqueue time
+		r.request.Headers = append(r.request.Headers, &httpgrpc.Header{
+			Key:    string(lokihttpreq.QueryEnqueueTimeHTTPHeader),
```

Comment:
nit:
I think it's a bit more correct to set/get the header via [CanonicalMIMEHeaderKey](https://pkg.go.dev/net/textproto#CanonicalMIMEHeaderKey). This is handled for us when getting via `req.Header.Get()`, but it doesn't look like there are `httpgrpc` utilities for setting it, so we can probably do it manually.

Note: your headers are already formatted correctly, but using these utilities will give us more confidence it stays this way as the codebase grows.
```suggestion
			Key:    textproto.CanonicalMIMEHeaderKey(string(lokihttpreq.QueryEnqueueTimeHTTPHeader)),
```

Reply:
Good point. Addressed in https://github.com/grafana/loki/pull/4949/commits/e54455730662767e894e453bfc5b2e75647e5dbe